edocumentation-domain/src/main/java/org/dcsa/edocumentation/domain/persistence/entity/ShippingInstruction.java

```diff
@@ -224,6 +224,20 @@ public class ShippingInstruction extends AbstractStateMachine<EblDocumentStatus>
   @JoinColumn(name = "shipping_instruction_id")
   private List<@Valid AdvanceManifestFilingEBL> advanceManifestFilings;
 
+  @ElementCollection
+  @Column(name = "requested_certificate", nullable = false)
+  @CollectionTable(name = "requested_carrier_certificate", joinColumns = @JoinColumn(name = "shipping_instruction_id"))
+  @RequestedCarrierCertificateValidation( groups = AsyncShipperProvidedDataValidation.class)
```

Comment:
Couldn't we just use `@PseudoEnum` here?

Reply:
I tried and it didn't work. Because we wanted to have the list of strings to be checked. I tried various ways. Hence I used validators.